pkg/build/package.go

```diff
@@ -243,6 +243,14 @@ func (pc *PackageBuild) generateControlSection(ctx context.Context) ([]byte, err
 		return nil, fmt.Errorf("unable to build control FS: %w", err)
 	}
 
+	locked, err := json.Marshal(pc.Build.resolvedApkoConfig)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal apko config: %w", err)
+	}
+	if err := fsys.WriteFile(".apko.lock.json", locked, 0644); err != nil {
```

Comment:
(i know this is a bike-shed, but...) WDYT about `.LOCK.apko.json` of `.BUILDLOCK.apko.json` 🤔 

My thinking is that we use `.apko.yaml` elsewhere to indicate things are the apko dialect of yaml.  Here it's json, but similar idea.  The "build" qualifier is so it's clearer what it's the locked apko configuration of.